packages/entity/src/Entity.ts

```diff
@@ -331,8 +331,13 @@ export interface IEntityClass<
   >,
   TSelectedFields extends keyof TFields = keyof TFields
 > {
-  new (viewerContext: TViewerContext, obj: Readonly<TFields>): TEntity;
-  getCompanionDefinition(): EntityCompanionDefinition<
+  new (constructorParam: {
+    viewerContext: TViewerContext;
+    id: TID;
+    databaseFields: Readonly<TFields>;
+    selectedFields: Readonly<Pick<TFields, TSelectedFields>>;
+  }): TEntity;
+  defineCompanionDefinition(): EntityCompanionDefinition<
```

Comment:
Consider documenting this method to explain that it is called once and memoized by the Entity framework.